erts/emulator/beam/jit/arm/ops.tab

```diff
@@ -1161,34 +1161,42 @@ gc_bif2 Fail Live u$bif:erlang:div/2 S1 S2 Dst => \
   i_m_div Fail Live S1 S2 Dst
 
 # Fused 'rem'/'div' pair.
-gc_bif2 Fail Live u$bif:erlang:rem/2 LHS RHS Remainder | \
-    gc_bif2 A B u$bif:erlang:intdiv/2 LHS RHS Quotient | \
-    distinct(LHS, Remainder) | \
-    distinct(RHS, Remainder) => \
-      i_rem_div Fail Live LHS RHS Remainder Quotient
+gc_bif2 Fail Live u$bif:erlang:rem/2 LHS1 RHS1 Remainder | \
+    gc_bif2 A B u$bif:erlang:intdiv/2 LHS2 RHS2 Quotient | \
+    equal(LHS1, LHS2) | \
```

Comment:
Given the changes to the types, would it make sense to not allow matching on equal arguments by repeating the pattern and always require explicit `equal` calls? This seems like something that would be easy to get wrong and might result in considerable performance issues

Reply:
Yes, I have also had the same thought. It is a probably a good idea.